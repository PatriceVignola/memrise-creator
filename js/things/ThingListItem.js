/* @flow */

import React from 'react';
import { ListItem } from 'react-native-paper';
import { withHandlers } from 'recompose';

import type { Thing } from '../actions/types';

type HigherOrderProps = {
  onPress: (thing: Thing) => void,
  thing: Thing,
};

type Props = {
  onPress: () => void,
  thing: Thing,
};

function ThingListItem(props: Props) {
  // TODO: Add an icon or image, maybe the first google image of the combination of both words?
  return (
    <ListItem
      onPress={props.onPress}
      title={props.thing.target.word}
      description={props.thing.source.word}
    />
  );
}

const enhance = withHandlers({
  onPress: (props: HigherOrderProps) => () => {
    props.onPress(props.thing);
  },
});

export default enhance(ThingListItem);
