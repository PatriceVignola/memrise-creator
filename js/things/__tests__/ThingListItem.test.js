/* @flow */

import React from 'react';
import { ListItem } from 'react-native-paper';
import Renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';

import ThingListItem from '../ThingListItem';

const shallowRenderer = new ShallowRenderer();

describe('ThingListItem', () => {
  const thingLanguage = {
    word: 'abc',
    kind: 'text',
    alts: ['def'],
  };

  const thing = {
    id: 1,
    source: thingLanguage,
    target: thingLanguage,
  };

  test('renders a thing', () => {
    const shallow = shallowRenderer.render((
      <ThingListItem
        onPress={jest.fn()}
        thing={thing}
      />
    ));

    expect(shallow).toMatchSnapshot();
  });

  test('passes the thing to its container when getting clicked', () => {
    const mockPressFunction = jest.fn();

    const deep = Renderer.create((
      <ThingListItem
        onPress={mockPressFunction}
        thing={thing}
      />
    ));

    deep.root.findByType(ListItem).props.onPress(thing);
    expect(mockPressFunction).toHaveBeenCalledTimes(1);
    expect(mockPressFunction).toHaveBeenCalledWith(thing);
  });
});
