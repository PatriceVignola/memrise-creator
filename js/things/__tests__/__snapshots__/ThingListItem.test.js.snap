// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThingListItem renders a thing 1`] = `
<ThingListItem
  onPress={[Function]}
  thing={
    Object {
      "id": 1,
      "source": Object {
        "alts": Array [
          "def",
        ],
        "kind": "text",
        "word": "abc",
      },
      "target": Object {
        "alts": Array [
          "def",
        ],
        "kind": "text",
        "word": "abc",
      },
    }
  }
/>
`;
